tdrs-backend/tdpservice/search_indexes/management/commands/clean_and_reparse.py

```diff
@@ -0,0 +1,236 @@
+
+            log("Database backup complete.",
+                logger_context=log_context,
+                level='info')
+        except Exception as e:
+            log("Database backup FAILED. Clean and re-parse NOT executed. Database and Elastic are CONSISTENT!",
+                logger_context=log_context,
+                level='error')
+            raise e
+
+    def __handle_elastic(self, new_indices, delete_indices, log_context):
+        """Create new Elastic indices and delete old ones."""
+        if new_indices:
+            try:
+                if not delete_indices:
+                    call_command('tdp_search_index', '--create', '-f', '--use-alias', '--use-alias-keep-index')
+                else:
+                    call_command('tdp_search_index', '--create', '-f', '--use-alias')
+                log("Index creation complete.",
+                    logger_context=log_context,
+                    level='info')
+            except Exception as e:
+                log("Elastic index creation FAILED. Clean and re-parse NOT executed. "
+                    "Database is CONSISTENT, Elastic is INCONSISTENT!",
+                    logger_context=log_context,
+                    level='error')
+                raise e
+            # TODO: Need to ask Alex if we can run some queries in prod to deduce the average number of records per DF.
+
+    def __delete_records(self, docs, file_ids, new_indices, log_context):
+        """Delete records and documents from Postgres and Elastic respectively."""
+        total_deleted = 0
+        for doc in docs:
+            # atomic delete?
+            try:
+                model = doc.Django.model
+                qset = model.objects.all().filter(datafile_id__in=file_ids)
+                total_deleted += qset.count()
+                if not new_indices:
+                    # If we aren't creating new indices, then we don't want duplicate data in the existing indices.
+                    doc().update(qset, refresh=True, action='delete')
+                qset._raw_delete(qset.db)
+            except BulkIndexError as e:
+                log(f'Elastic document delete failed for model {model}. Database and Elastic are INCONSISTENT! Restore '
+                    'the DB from the backup as soon as possible!',
+                    logger_context=log_context,
+                    level='critical')
+                raise e
+            except Exception as e:
+                log(f'_raw_delete failed for model {model}. Database and Elastic are INCONSISTENT! Restore the DB from '
+                    'the backup as soon as possible!',
+                    logger_context=log_context,
+                    level='critical')
+                raise e
+        return total_deleted
+
+    def __handle_datafiles(self, files, log_context):
+        """Delete, re-save, and re-parse selected datafiles."""
+        for f in files:
+            try:
```

Comment:
you can put both delete and save in one try/catch and catch two separate exceptions